salt/modules/zypper.py

```diff
@@ -1308,7 +1308,10 @@ def list_products(all=False, refresh=False):
 
     ret = list()
     OEM_PATH = "/var/lib/suseRegister/OEM"
-    cmd = _zypper('-x', 'products')
+    cmd = _zypper()
+    if not all:
+        cmd.append('--disable-repos')
+    cmd.extend(['-x', 'products'])
```

Comment:
I think `--disable-repos` is a global option, and therefore has to go before `products`